src/dgipy/vcf.py

```diff
@@ -0,0 +1,202 @@
+
+import contextlib
+from itertools import groupby
+
+import pandas as pd
+import pysam
+import requests
+from tqdm import tqdm
+
+import dgipy
+
+# Sample usage: import vcf
+#               data = vcf.annotate('link/to/file',chr='chr#')
+
+
+# TODO: Probably need another class as a wrapper object rather than putting it all in a list
+# Class would have analogous display methods but also allow access to individual GeneResults
+class GeneResult:
+    """A gene result from original VCF
+
+    Attributes
+    ----------
+    gene : str
+        The name of the gene
+    records : list
+        The mapped records from original VCF
+    interactions : pd.DataFrame
+        The interactions obtained from DGIdb
+
+    Methods
+    -------
+    search_interactions(gene: str) -> pd.DataFrame
+        Search drug-gene interactions for given gene in DGIdb
+
+    """
+
+    def __init__(self, data: list) -> None:
+        """Initialize a gene with a name, VCF records, and drug-gene interactions"""
+        # TODO: handle genes without names, 'novel transript'
+        try:
+            self.gene = data[0]["name"]
+        except:
+            self.gene = "None"
+
+        self.records = data
+        self.interactions = self.search_interactions(self.gene)
+
+        # TODO: handle app searches with blank lists [], currently FDA resource hangs for awhile?
+        if not list(self.interactions["drug"].values):
+            self.applications = "None"
+        else:
+            self.applications = self.search_applications(
+                list(self.interactions["drug"].values)
+            )
+
+        self.gene_info = self.grab_gene_info(self.gene)
+        self.categories = self.grab_categories(self.gene)
+
+    # TODO: These can probably just be simpilifed to direct assignment during __init__ but might be useful this way too
+    def search_interactions(self, gene: str) -> pd.DataFrame:
+        """Search drug-gene interactions for given gene in DGIdb
+
+        :param gene: the name of the gene
+        :return: Dataframe of drug-gene interactions
+        """
+        return dgipy.get_interactions(gene)
+
+    def search_applications(self, drugs: list) -> pd.DataFrame:
+        """Search for drug applications for interaction drugs in DGIdb
+
+        :param drugs: list of drugs from interaction results
+        :return: Dataframe of FDA applications
+        """
+        return dgipy.get_drug_applications(drugs)
+
+    def grab_gene_info(self, gene: str) -> pd.DataFrame:
+        """Grab gene information from DGIdb
+
+        :param gene: gene name
+        :return: Dataframe of gene information
+        """
+        return dgipy.get_gene(gene)
+
+    def grab_categories(self, gene: str) -> pd.DataFrame:
+        """Grab gene categories from DGIdb
+
+        :param gene: gene name
+        :return: Dataframe of gene categories
+        """
+        return dgipy.get_categories(gene)
+
+
+def annotate(filepath: str, contig: str) -> pd.DataFrame:
```

Comment:
file args should be `pathlib.Path` -- see https://docs.python.org/3/library/pathlib.html

Reply:
added